retriever/__main__.py

```diff
@@ -63,7 +63,10 @@ def main():
             return
 
         if args.command == 'update':
-            check_for_updates()
+            if update_scripts():
+                print("\nThe retriever is up-to-date")
+            else:
+                raise
```

Comment:
Having this defensive logic here is a little confusing to read since `update_scripts()` is doing something not just checking something. My initial thought would be to keep this logic inside `update_scripts()` and have it `raise` or `print` from there. Is there a reason to do it this way instead?

I'm also not sure this will actually execute as expected since I think the `raise` in `update_scripts` will prevent the `return False` from executing (see my comment below).